app/controllers/links_controller.rb

```diff
@@ -8,10 +8,7 @@ def new
 
   def index
     @links = Link.all
-    @popular = Link.where("clicks >= ?", 1).order(clicks: :desc)
-    @recent = Link.order(created_at: :desc)
-    @top_users = User.top_users
-    @user_links = current_user.links.order(created_at: :desc) if logged_in?
+    @presenter = Links::IndexPresenter.new
```

Comment:
**Code Quality-Critical**
Move view logic to presenter. Rails convention advices against having many intance variables in one controller action. This works to enforce the `single responsibility principle`.